# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalRequest.h
# Commit: a1fcf22a2a0e
# Full Hash: a1fcf22a2a0e003f83a6f66863cc96805ba27a2a
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-12 21:30:33
# Regressor Bug: 1351231
# File Overlap Count: 1
# Description:
#   Bug 1351231 - Support conversation between InternalRequest and IPCInternalRequest. r=dom-worker-reviewers,jesup
#   
#   Depends on D138812
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138813
# ==============================================================================

diff -r fd2a843599d1 -r a1fcf22a2a0e dom/fetch/InternalRequest.h
--- a/dom/fetch/InternalRequest.h	Thu Jan 12 17:24:27 2023 +0000
+++ b/dom/fetch/InternalRequest.h	Thu Jan 12 17:24:27 2023 +0000
@@ -29,6 +29,7 @@
 namespace mozilla {
 
 namespace ipc {
+class PBackgroundChild;
 class PrincipalInfo;
 }  // namespace ipc
 
@@ -97,6 +98,9 @@
 
   explicit InternalRequest(const IPCInternalRequest& aIPCRequest);
 
+  void ToIPCInternalRequest(IPCInternalRequest* aIPCRequest,
+                            mozilla::ipc::PBackgroundChild* aManager);
+
   SafeRefPtr<InternalRequest> Clone();
 
   void GetMethod(nsCString& aMethod) const { aMethod.Assign(mMethod); }